base-permissions/src/commonMain/kotlin/Permission.kt

```diff
@@ -89,21 +90,24 @@ open class PermissionsBuilder(val context: PermissionContext = defaultPermission
         } ?: throw Error("Permission state repo factory was not registered for $permission")
 }
 
+private val defaultPermissionDispatcher by lazy {
```

Comment:
unsure on this approach. I'd prefer to remove the default value completely, but I'm not sure how many clients are relying on it. in this case one thread would be created if needed, however it'd live forever

Reply:
Thread are not super heavy leaks even though they are not trivial either (essentially one stack size) so it should be ok. Good idea to make it `lazy` though.

For the record, with `creator` + `native-mt` it's still easy to have a simple pool if you'd prefer, you can just make a group of singleThreadDispatchers and hand them out round robin. This doesn't redistribute load, but does limit your threads while not grouping everything in one single thread.

But IMHO it's not better than what we will have with thread pool context though, so you will probably agree to keep things as is.

Reply:
I think it'd be ok for now, but I have high hopes for the next version of the coroutines with `limitedParallelism`. interesting idea on the pool, however handling exceptions might be rather tricky..